ruby_weather.rb

```diff
@@ -0,0 +1,98 @@
+require 'faraday'
+require 'json'
+require 'uri'
+
+#Get Postal-Code from User
+def get_postal_code(prompt)
+  puts prompt
+  postal_code = gets.chomp().to_s
+end
+
+#Validate the Postal-Code
+def validate_postal_code(postal_code, no_more_tries, count, limit, invalid,prompt, tries)
+  while postal_code.length != 6 and !no_more_tries
+    if count < limit
+      puts invalid
+      puts prompt
+      postal_code = gets.chomp().to_s
+      count += 1
+    else
+      no_more_tries = true
+    end
+  end
+
+  if no_more_tries
+    puts tries
+    return
+  end
+end
+
+#Get the Location Key
+def api_call_1(api_key, params, key_url)
+
+  res = Faraday.get(key_url) do |req|
+    req.params['apikey'] = api_key
+    req.params['q'] = params
+
+  end
+  resp = JSON.parse(res.body)
+end
+
+#Get the Current Conditions
+def api_call_2(api_key, params, key_url)
+
+  res = Faraday.get(key_url) do |req|
+    req.params['apikey'] = api_key
+    req.params['details'] = params
+
+  end
+  resp = JSON.parse(res.body)
+end
+
+#output the Weather Data
+def output_weather(resp)
+  condition = resp[0]["WeatherText"]
+  temperature = resp[0]["Temperature"]["Metric"]["Value"].to_s
+  feels_like = resp[0]["RealFeelTemperature"]["Metric"]["Value"].to_s
+
+  puts "*************************************"
+  puts "* Current Condition is " + condition
+  puts "* Temperature is " + temperature + "C"
+  puts "* Feels Like temperature is " + feels_like + "C"
+  puts "*************************************"
+
+end
+
+#Call all methods and get the Weather Output
+def get_weather()
+  prompt = "> Enter your Postal Code (A1A1A1)"
+  invalid = "***** INVALID POSTAL CODE ****** \n "
+  tries = "Number of tries exceeded"
+  count = 0
+  limit = 2
+  no_more_tries = false
+  key_url = "http://dataservice.accuweather.com/locations/v1/postalcodes/CA/search"
+  api_key = "TCM86A7nzGw6RDfuWLJFhO0Q6VF5Vhdl"
```

Comment:
This isn't particularly sensitive given the use, but generally you don't want to store your API keys as plain text in your code as it means anyone can read them. They should be treated like passwords. 

A better way to handle this would be via an Environment Variable. You would define that in your terminal, and Ruby could pull that value in when you run your code. 

https://www.rubyguides.com/2019/01/ruby-environment-variables/